cmd/mesg-cli/orchestrator.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/crypto/keys"
+	"github.com/mesg-foundation/engine/cosmos"
+	"github.com/mesg-foundation/engine/event/publisher"
+	"github.com/mesg-foundation/engine/ext/xsignal"
+	"github.com/mesg-foundation/engine/orchestrator"
+	"github.com/mesg-foundation/engine/server/grpc"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+func orchestratorCmd(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "orchestrator",
+		Short: "Orchestrator subcommands",
+	}
+	cmd.AddCommand(flags.GetCommands(
+		startOrchestratorCmd(cdc),
+	)...)
+	return cmd
+}
+
+func startOrchestratorCmd(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Start the Orchestrator",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			if viper.GetString(flagMnemonic) == "" {
+				return fmt.Errorf("mnemonic is required. use flag --mnemonic or config file")
+			}
+			if cliCtx.ChainID == "" {
+				return fmt.Errorf("chain-id is required. use flag --chain-id or config file")
+			}
+
+			logger := log.NewTMLogger(log.NewSyncWriter(os.Stdout)).With("module", "orchestrator")
+			client, err := cliCtx.GetNode()
+			if err != nil {
+				return err
+			}
+
+			// init rpc client
+			logger.Info("starting rpc client")
+			if err := client.Start(); err != nil {
+				return err
+			}
+			defer func() {
+				logger.Info("stopping rpc client")
+				if err := client.Stop(); err != nil {
+					logger.Error(err.Error())
+				}
+			}()
+
+			kb := cosmos.NewInMemoryKeybase()
+			if _, err := kb.CreateAccount("orchestrator", viper.GetString(flagMnemonic), "", "password", keys.CreateHDPath(0, 0).String(), cosmos.DefaultAlgo); err != nil {
```

Comment:
Might be good to have the `accName`, `password`, `accNumber` and `accIndex` from the flags or a const

Reply:
- `accName` and `password` are only used internally
- but yes it could be useful to have `accNumber` and `accIndex` 👍

Reply:
yes so just as a const might be good, it took me quite some time to figure out why we put "password" in plain text. Just for code quality to avoid to have constants but not a big deal here

Reply:
done